src/runtime_src/core/tools/xbmgmt2/OO_Hotplug.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (C) 2022 AMD, Inc
+ * 
+ * Licensed under the Apache License, Version
+ * 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. A copy of the License is located
+ * at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_Hotplug.h"
+#include "tools/common/XBUtilitiesCore.h"
+#include "tools/common/XBUtilities.h"
+
+// XRT - Include Files
+#include "core/common/query_requests.h"
+#include "core/common/system.h"
+
+// 3rd Party Library - Include Files
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <fstream>
+// =============================================================================
+
+// ----- H E L P E R M E T H O D S ------------------------------------------
+static void
+hotplug_online()
+{
+  static const std::string  rescan_path = "/sys/bus/pci/rescan";
+
+  if (!boost::filesystem::exists(rescan_path))
+    throw xrt_core::error((boost::format("Invalid sysfs file path '%s'.") % rescan_path).str());
+
+  std::ofstream rescan_file(rescan_path, std::ofstream::out);
+  try
```

Comment:
Why is there a try here if you are re-throwing the error? I don't think it's needed here

Reply:
Resolved. Thanks